### PR TITLE
linux support

### LINKED ISSUE #9: Linux can't find device (presumably libusb/hidapi)
Reports from atsamd gitter that linux flashing isnt finding devices

we could more explicitly link the linux deps https://github.com/libusb/hidapi#linux

but even that hasnt seemed to help

> @jacobrosenthal Ubuntu mate 18.04, I had libusb-1.0.0 already installed but just installed libudev and libfox as per the libusb/hidapi install instructions. I'll update if it works when I get in front of my device in a few hours
> I've only just installed those deps and force reinstalled cargo-hf2 (not sure if there's something else I need to do). I'm still getting the same error message with these logs: https://gist.github.com/tekulvw/d9db07633c213145c093c2b1668a932b
>

### PR DESCRIPTION
**Pull request by jacobrosenthal**
closes #9

### CODE REVIEW

**Comment by jacobrosenthal** [comment]
Also requires #14

**Comment by jacobrosenthal** [comment]
TLDR  linux users Is this working now? You can cargo install from this branch with
`cargo install --git https://github.com/jacobrosenthal/hf2-rs --branch bininfo --force cargo-hf2`

---

The first problem with linux support is frankly documentation and deps. Weve seen something like the attached instructions get a few people there, but we could use more on that front.

Then we started seeing the second packet in the checksums command in `rx()` get 0 bytes for the nth packet. But the first packet and other commands with single packet result are working.

All the wiresharking didn't clear anything up, the command being sent, and the first packet received looked fine, we just werent getting the second packet.

After all our debugging we noticed the upstream code actually does work like 1/10 times! So there is technically nothing wrong upstream code. The best we can figure is to put 5 retries on the rx, and were now seeing successful uploads in linux. 

I could see maybe only retrying if its a second packet? or maybe just sleeping for some amount of time and maybe no retries? But this set of commits works so Id like some more feedback from linux users.
